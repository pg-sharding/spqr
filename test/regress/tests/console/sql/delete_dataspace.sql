CREATE DATASPACE ds1;
CREATE DATASPACE ds2;

CREATE SHARDING RULE rule1 COLUMNS id FOR DATASPACE ds1;
CREATE SHARDING RULE rule2 COLUMNS id FOR DATASPACE ds2;

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DATASPACE ds1;
CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh2 FOR DATASPACE ds1;

CREATE KEY RANGE krid3 FROM 11 ROUTE TO sh2 FOR DATASPACE ds2;

-- drop non-existing one
DROP DATASPACE ds23223;

-- non-cascade drop
DROP DATASPACE ds1;

DROP DATASPACE ds1 CASCADE;

SHOW dataspaces;

DROP DATASPACE ALL CASCADE;

SHOW dataspaces;
SHOW key_ranges;
SHOW sharding_rules;