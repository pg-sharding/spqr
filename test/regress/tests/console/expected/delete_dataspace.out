
		SQPR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DATASPACE ds1;
         add dataspace         
-------------------------------
 created dataspace with id ds1
(1 row)

CREATE DATASPACE ds2;
         add dataspace         
-------------------------------
 created dataspace with id ds2
(1 row)

CREATE SHARDING RULE rule1 COLUMNS id FOR DATASPACE ds1;
                                    add sharding rule                                     
------------------------------------------------------------------------------------------
 created sharding rule rule1 for table (*) with columns [id, hash: x->x] in ds1 dataspace
(1 row)

CREATE SHARDING RULE rule2 COLUMNS id FOR DATASPACE ds2;
                                    add sharding rule                                     
------------------------------------------------------------------------------------------
 created sharding rule rule2 for table (*) with columns [id, hash: x->x] in ds2 dataspace
(1 row)

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DATASPACE ds1;
         add key range          
--------------------------------
 created key range with bound 1
(1 row)

CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh2 FOR DATASPACE ds1;
          add key range          
---------------------------------
 created key range with bound 11
(1 row)

CREATE KEY RANGE krid3 FROM 11 ROUTE TO sh2 FOR DATASPACE ds2;
          add key range          
---------------------------------
 created key range with bound 11
(1 row)

-- drop non-existing one
DROP DATASPACE ds23223;
 drop dataspace 
----------------
(0 rows)

-- non-cascade drop
DROP DATASPACE ds1;
ERROR:  cannot drop dataspace ds1 because other objects depend on it
HINT: Use DROP ... CASCADE to drop the dependent objects too.
DROP DATASPACE ds1 CASCADE;
   drop dataspace   
--------------------
 drop dataspace ds1
(1 row)

SHOW dataspaces;
 Dataspace ID 
--------------
 default
 ds2
(2 rows)

DROP DATASPACE ALL CASCADE;
   drop dataspace   
--------------------
 drop dataspace ds2
(1 row)

SHOW dataspaces;
 Dataspace ID 
--------------
 default
(1 row)

SHOW key_ranges;
 Key range ID | Shard ID | Dataspace ID | Lower bound 
--------------+----------+--------------+-------------
(0 rows)

SHOW sharding_rules;
 Sharding Rule ID | Dataspace ID | Table Name | Columns | Hash Function 
------------------+--------------+------------+---------+---------------
(0 rows)

